plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/OTelTelemetryPlugin.java

```diff
@@ -68,10 +78,31 @@ public String getName() {
 
     private Telemetry telemetry(TelemetrySettings telemetrySettings) {
         final OpenTelemetrySdk openTelemetry = OTelResourceProvider.get(telemetrySettings, settings);
+        final RefCountedReleasable<SdkTracerProvider> refCountedSdkTracerProvider = new RefCountedReleasable<>(
```

Comment:
I really don't understand what you are doing here (my apologies if I am missing something):
 - `SdkTracerProvider` is not shareable - it is used only by `OTelTracingTelemetry` (and could be closed by this class)
 - `SdkMeterProvider` is not shareable - it is used only by `OTelMetricsTelemetry` (and could be closed by this class)

What is shareable is `OpenTelemetrySdk` instance that should be closed by either `OTelMetricsTelemetry`  or OTelTracingTelemetry` . The problem is that the feature flags you are trying to introduce does not play at all with the way telemetry is initialized: it is eager for both tracing and metrics even if it is no need at all. 

The way we could solve that with at least two options:
 - drop `Telemetry` completely and use two extension points instead:
   ```
@ExperimentalApi
public interface TelemetryPlugin {

    /**
     * Provides tracing telemetry
     * @return tracing telemetry instance
     */
    Optional<TracingTelemetry> getTracingTelemetry();

    /**
     * Provides metrics telemetry
     * @return metrics telemetry instance
     */
    Optional<MetricsTelemetry> getMetricsTelemetry();

    String getName();
}
```

 - initialize `OpenTelemetrySdk`  once (and close it in `close()` method) but let's `OTelTelemetry` initialize `OTelTracingTelemetry` and `OTelMetricsTelemetry` lazily instead, in this case each of them could increment the reference so it won't be closed prematurely

Reply:
I think we can go ahead with approach 2 for now. A lot of things like settings and all are associated with telemetry and it would be a bigger refactoring.